src/Command/CommandHelper.php

```diff
@@ -96,7 +96,7 @@ public static function begin(
 		if ($allowXdebug && !XdebugHandler::isXdebugActive()) {
 			$errorOutput->getStyle()->note('You are running with "--xdebug" enabled, but the Xdebug PHP extension is not active. The process will not halt at breakpoints.');
 		} elseif (!$allowXdebug && XdebugHandler::isXdebugActive()) {
-			$errorOutput->getStyle()->note('The Xdebug PHP extension is active, but "--xdebug" is not used. This may slow down performance and the process will not halt at breakpoints.');
+			$errorOutput->getStyle()->note('The Xdebug PHP extension is active, but "--xdebug" is not used. Using "--xdebug" may slow down performance, but will allow you to debug PHPStan using breakpoints.');
```

Comment:
the point of this message is, that php processes may slow down because xdebug beeing loaded.

phpstan doesn't get slowser because of `--xdebug`

Reply:
Doesn't xdebug handler prevent that by disabling the extension?

My understanding of --xdebug is that it prevents xdebug handler from doing so.

Reply:
When xdebug is loaded things get a lot slower - e.g. when running `make tests`.

If you don't run with `--xdebug` you get a slow process without any benefits - hence the warning.

Reply:
The whole point of `XdebugHandler` is to restart the process without Xdebug exactly to avoid this slow down, even if it is installed. By testing for `XdebugHandler::isXdebugActive()` before `$xdebug->check();` this message is being shown when it's not relevant.

Here are some numbers from a fairly large project.
php phpstan: 117 sec
php+xdebug phpstan: 110 sec
php+xdebug phpstan --xdebug: 264 sec

Maybe you had `PHPSTAN_ALLOW_XDEBUG=1` set in your enviroment when testing this so that Xdebug was left active? This would cause Xdebug to still be loaded and result in the behavior you describe (https://github.com/composer/xdebug-handler#basic-usage).

Reply:
I see the confusing nature of the message has been pointed out by others as well: https://github.com/phpstan/phpstan-src/pull/1878#issuecomment-1336982466

How about this phrasing instead?

> The Xdebug PHP extension is active. Consider disabling if not using "--xdebug" to avoid performance issues.

I took out `and the process will not halt at breakpoints` to keep the message relatively short as users are unlikely to have set any breakpoints if they where unaware that Xdebug is loaded.

If you don't find either of these versions are good then could you please help rephrase it since it sounds like it's mostly related to how `make tests` works internally, something I'm not familiar with, yet it's causing a lot of confusion in it's current form (I see people setting up there CI to run slower because of miss understanding the message, thinking that `--xdebug` will improve performance when Xdebug is active).